src/wp-includes/media.php

```diff
@@ -2072,7 +2100,7 @@ function wp_img_tag_add_decoding_attr( $image, $context ) {
 	 * @param string            $context Additional context about how the function was called
 	 *                                   or where the img tag is.
 	 */
-	$value = apply_filters( 'wp_img_tag_add_decoding_attr', 'async', $image, $context );
+	$value = apply_filters_deprecated( 'wp_img_tag_add_decoding_attr', 'async', $image, $context );
```

Comment:
It would be nice to be able to continue to support this filter for back-compat purposes unless we've already determined that it's not being used. Perhaps instead of the new filter for `wp_decoding_value` that is added below?

Reply:
@joemcgill 

Great point. What would make the most sense here?

1. Rename `wp_decoding_value` and keep `wp_img_tag_add_decoding_attr`— though semantically attached to a deprecated function;
2. Apply `wp_img_tag_add_decoding_attr` just before applying `wp_decoding_value` when getting `wp_get_loading_optimization_attributes()`;

I am inclined to the 2nd option.